app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -0,0 +1,135 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::LiveMetricsCapture
+    class TargetValidationError < RuntimeError; end
+    class MetricValidationError < RuntimeError; end
+
+    def initialize(target)
+      @target = target
+      unless @target.kind_of?(MiddlewareServer)
+        raise TargetValidationError, "Validation error: unknown target"
+      end
+      @ems = @target.ext_management_system
+    end
+
+    def metrics_available
+      resource = Struct.new(:id, :feed).new
+      resource.id = @target.nativeid
+      resource.feed = @target.feed
+      available = []
+      @ems.metrics_resource(resource).each do |metric|
+        available << {
+          :id   => metric.id,
+          :name => @target.class::METRICS_HWK_MIQ[metric.name],
+          :type => metric.type,
+          :unit => metric.unit
+        }
+      end
+      available
```

Comment:
iteration with `.each` with a `<<` inside can be simplified to `.collect`

So you can drop the `available` variable and just do:

```ruby
      @ems.metrics_resource(resource).collect do |metric|
        {
          :id   => metric.id,
          :name => @target.class::METRICS_HWK_MIQ[metric.name],
          :type => metric.type,
          :unit => metric.unit
        }
      end
```